### PR TITLE
Includes `download_type` parameter to select the type of download job

### PR DESCRIPTION
**Pull request by apozsuse**
Renamed the file `mash/services/download/obs_build_result.py` to`obs_job.py` as Sean suggested. There only the class name has been changed. Additionally changed the old `obs_job.py` to `service.py`. 

That change in the names messed up with the diff :face_with_spiral_eyes: it might be easier to check the files directly.

Highlighting the changes in each file in comments...

### CODE REVIEW

**Comment by apozsuse** [comment]
Changes in `mash/services/download/service.py`:
- Change the name of the class to `DownloadService`.
- Lines 248-251 includes `download_type` parameter to instantiate the approrpiate class:
```python
        if 'download_type' in job and job['download_type'] == 'S3':
            job_worker = S3DownloadJob(**kwargs)
        else:
            job_worker = OBSDownloadJob(**kwargs)
```

**Comment by apozsuse** [comment]
Changes in `mash/services/download/obs_job.py`:
- Changes the class name to `OBSDownloadJob`

**Review by rjschwei — CHANGES_REQUESTED** [review]

File: mash/services/api/v1/schema/jobs/__init__.py [hunk 1]
```diff
@@ -128,6 +128,15 @@
                            'itself. Valid condition operators are >, <, >=, '
                            '<=, or ==.'
         },
+        'download_type': string_with_example(
+            'OBS',
+            description='The type of download to get the source image file.'
```
[the remark this answers is not in the document]
Comment (apozsuse) [reply mash/services/api/v1/schema/jobs/__init__.py]: Fixed!

File: mash/services/api/v1/schema/jobs/__init__.py [hunk 2]
```diff
@@ -128,6 +128,15 @@
...
+                        'Supported values are:'
+                        '- `OBS` (default) when the image has to be downloaded'
+                        ' from a OpenBuildService instance.'
+                        '- `S3` when the image file to be processed is stored'
+                        ' in a AWS S3 bucket. '
```
[the remark this answers is not in the document]
Comment (apozsuse) [reply mash/services/api/v1/schema/jobs/__init__.py]: Fixed

**Review by smarlowucf — APPROVED** [review]

**Pull request merged**